### PR TITLE
Improve external catalog search

### PR DESCRIPTION
**Pull request by bpfoley**

### CODE REVIEW

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:180]: Why the commented code?

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:116]: Leftover debugging?

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:211]: Leftover debugging?

**Review by cpeel — COMMENTED** [review]
We should get a sandbox spun up for this as well I think.

**Comment by bpfoley** [comment]
> We should get a sandbox spun up for this as well I think.

Yes. It's at https://www.pgdp.org/~bfoley/c.branch/task1900-better-yaz/

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:211]: Yeah, deliberately, since it's slightly non-obvious. But I can remove it.

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:116]: Feature request from Sharon in task 1900.

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:180]: I was initially trying to figure out how to express 'language material only' in the query to the LoC server which would avoid the need to filter the response,  but I was getting weird and inconsistent results.

The original was commented, and I was undecided if I wanted to enable it or not.

Then I thought I might use this logic to decorate each table entry with an HTML class so we can selectively show/hide them in the browser with some javascript.

So... I'm undecided what I want to do with this, basically.

File: tools/project_manager/external_catalog_search.php
Comment (srjfoo) [reply tools/project_manager/external_catalog_search.php:116]: Well, more or less, anyway. 😁 

If a PM creating a project has to wade 10 pages of results to find the record they want, it's a useful reminder. Most PMs are probably not going to know what `@attr 1=1003` means, but it's pretty easy to figure out.

I also really like seeing the media type the record is for. One of my tests, the first three or four hits were all either sound recordings or video of some kind.

Comment (cpeel) [reply tools/project_manager/external_catalog_search.php:116]: I defer to Sharon as someone who actually uses this. To me this feels like it will generate more confusion than value for the majority of our PMs.

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [reply tools/project_manager/external_catalog_search.php:211]: Perhaps instead add a comment in `MARCRecordTest.php` about how to generate the encoded data, eg:
```php
// To generate data for these test functions, use:
// base64_encode(serialize($rec)) where $rec is the output of yaz_record()
```

**Review by cpeel — APPROVED** [review]

File: tools/project_manager/external_catalog_search.php
Comment (srjfoo) [reply tools/project_manager/external_catalog_search.php:116]: If you want to hold off merging for a bit, I'll ask Rick what he thinks. If he doesn't like the idea, we can ditch it. I don't want to make things harder for PMs.

Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:116]: No rush. Let's figure out what we're all happy with.

One possible alternative would be to hide the raw query in a `<details>` element that's closed by default.

**Review by srjfoo — APPROVED** [review]

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:202]: Should we move some of this styling into CSS? I'd rather have labeled-snowflakes there than style dusted around the code.

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:208]: We should `html_safe($errorMsg)` because while it's probably OK we shouldn't risk it as it comes from an external source. I ack that this was in the original.

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:255]: `$url_base` is output in the HTML directly so we either need to `&` -> `&amp;` here, or in the output function -- the latter is probably preferred. Ditto below where we continue constructing it.

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:294]: re: FA icon - I like this. We should replace our custom "open in a new window" icon with this site-wide. Not in this PR, obvs. Probably should think about doing this for all or our little inline graphics.

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:323]: Ditto raw `&` in final HTML.

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:337]: This too, no?
```suggestion
    return $show_link ? "<a href='$url_base&start=$start'>" . html_safe($label) . "</a>" : html_safe($label);
```

File: tools/project_manager/external_catalog_search.php
Comment (cpeel) [inline tools/project_manager/external_catalog_search.php:410]: tip: using `sprintf()` would probably make this easier to read.

Comment (cpeel) [reply tools/project_manager/external_catalog_search.php:410]: Ditto the others in this function.

File: tools/project_manager/marc_inspector.php
Comment (bpfoley) [reply tools/project_manager/marc_inspector.php:24]: I deliberately didn't translate any of this because, with the exception of the leader (which, TBH, I should probably change back) they're taken verbatim from the titles in the MARC 21 spec.

Comment (cpeel) [reply tools/project_manager/marc_inspector.php:24]: Makes sense, let's add a comment to that effect please.

File: tools/project_manager/marc_inspector.php
Comment (bpfoley) [reply tools/project_manager/marc_inspector.php:8]: Done. I've no idea why I did that...

File: tools/project_manager/marc_inspector.php
Comment (bpfoley) [reply tools/project_manager/marc_inspector.php:118]: Nice! Done.

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:202]: Done, after a smaller adventure in learning about `lessc`.

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:255]: ACK. Sprinkled `attr_safe` where `$url_base` is used. Also I'd left an accidental (and now unnecessary) `if ($hide_nontext) { $url_base .= "&hide_nontext=on"; }` in the code.

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:294]: Thanks. Ray pointed me in the right direction. The minor niggle I have about this is I don't like the size. I couldn't figure out how to make it size proportional to the `font-size` of the `div`.

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:323]: Handled along with the fix above.

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:337]: Yup! :)

File: tools/project_manager/external_catalog_search.php
Comment (bpfoley) [reply tools/project_manager/external_catalog_search.php:410]: Now that you say it, it probably would! Done.

File: tools/project_manager/marc_inspector.php
Comment (bpfoley) [reply tools/project_manager/marc_inspector.php:24]: Done.

File: styles/layout.less
Comment (cpeel) [inline styles/layout.less:359]: We should try not to hard-code color in `layout.less` because it may not work for all themes. See if one of the pre-defined `@border-color-base*` values will work for what you want as the themes override these.

Comment (bpfoley) [reply styles/layout.less]: ACK. Yay for CSS variables.

**Review by cpeel — APPROVED** [review]
LGTM. @srjfoo - any last review/thoughts re CSS before I merge this?

File: tools/project_manager/marc_inspector.php
Comment (srjfoo) [reply tools/project_manager/marc_inspector.php]: Agreed. Not everything needs to be translated,

**Review by srjfoo — APPROVED** [review]
Changes look good to me. For those who really like going through everything, they can still get the full results, but this allows a lot more flexibility than we had before.

**Pull request merged**
